crowbar_framework/lib/upgrade.rb

```diff
@@ -0,0 +1,62 @@
+class Upgrade
```

Comment:
Yes, I was thinking more about accessing specific parts, not the whole map. But it might be good, yes